FSQLocationBrokerTests/FSQCurrentLocationTests.m

```diff
@@ -0,0 +1,175 @@
+    
+    NSArray<CLLocation *> *locations2 = @[self.location2];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations2];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location2.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+    
+    NSArray<CLLocation *> *locations3 = @[self.location3];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations3];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+}
+
+- (void)testCurrentLocationWithOutOfOrderDatesMultipleCalls  {
+    NSComparisonResult result;
+
+    NSArray<CLLocation *> *locations2 = @[self.location2];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations2];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location2.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+    
+    NSArray<CLLocation *> *locations3 = @[self.location3];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations3];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+    
+    NSArray<CLLocation *> *locations1 = @[self.location1];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations1];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+}
+
+- (void)testCurrentLocationWithReverseDatesMultipleCalls {
+    NSComparisonResult result;
+
+    NSArray<CLLocation *> *locations3 = @[self.location3];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations3];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+    
+    NSArray<CLLocation *> *locations2 = @[self.location2];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations2];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+    
+    NSArray<CLLocation *> *locations1 = @[self.location1];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations1];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+}
+
+- (void)testCurrentLocationWithSameTimestamp {
```

Comment:
Yes, it's better to have a deterministic way of knowing what current location will be when timestamps are equal. I added a comment that stating first location processed wins.